nixos-module.nix

```diff
@@ -105,6 +110,15 @@ in
         virtualisation.docker.enable = true;
         virtualisation.arion.docker.client.package = pkgs.docker;
       })
+      (mkIf (cfg.backend == "docker-rootless") {
+        virtualisation = {
+          docker.rootless = {
+            enable = true;
+            setSocketVariable = true;
```

Comment:
`setSocketVariable` is not needed, because it does not affect systemd units. Systemd unit environment variables are independent.